src/java.base/share/classes/java/lang/Throwable.java

```diff
@@ -767,42 +767,30 @@ public void printStackTrace(PrintWriter s) {
      * Wrapper class for PrintStream and PrintWriter to enable a single
      * implementation of printStackTrace.
      */
-    private abstract static class PrintStreamOrWriter {
+    private interface PrintStreamOrWriter {
```

Comment:
This is still too many classes. Can we just pull out `println` to a method like:
```java
private static void println(Object printer, Object o) {
    if (printer instanceof printStream ps) {
        ps.println(o);
    } else {
        ((PrintWriter) printer).println(o);
    }
}
```
Should be fewer classes to accomplish the same functionality.